### PR TITLE
Feat: Smiley counter page and widget (kids-1271)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Introduced `AddMemberCounterPage` for users to specify the number of family members when adding them.
	- Added `SmileyCounter` widget to visually represent the number of family members using smiley icons.
	- Enhanced `FamilyOverviewPage` to dynamically change the app bar title based on the state of family members.
	- New `NoChildrenPage` layout improves responsiveness and simplifies the interface for adding members.
	- Introduced `maxAmount` parameter in `AmountCounter` for flexible amount constraints.
	- Added `smilePurpleIcon` widget to the icon library for enhanced UI options.

- **Bug Fixes**
	- Improved navigation logic in `FamilyOverviewPage` to handle results more robustly.

- **Style**
	- Updated UI components for better text rendering and layout consistency.

### CODE REVIEW

File: lib/features/children/overview/pages/family_overview_page.dart [hunk 1]
```diff
@@ -90,14 +95,19 @@ class FamilyOverviewPage extends StatelessWidget {
...
   }
 
-  void _addNewChild(BuildContext context, FamilyOverviewUpdatedState state) {
-    AnalyticsHelper.logEvent(
+  void _addNewChild(
+      BuildContext context, FamilyOverviewUpdatedState state) async {
+    unawaited(AnalyticsHelper.logEvent(
       eventName: AmplitudeEvents.addMemerClicked,
+    ));
+    final dynamic result = await Navigator.push(
```
Comment (TammiLion) [inline lib/features/children/overview/pages/family_overview_page.dart:103]: I'm assuming this is test code that you don't want to merge yet?

File: lib/features/children/add_member/pages/add_member_counter_page.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:givt_app/features/children/add_member/widgets/smiley_counter.dart';
+import 'package:givt_app/features/children/generosity_challenge/widgets/generosity_app_bar.dart';
+import 'package:givt_app/features/children/generosity_challenge/widgets/generosity_back_button.dart';
+import 'package:givt_app/features/family/shared/widgets/content/amount_counter.dart';
+import 'package:givt_app/features/family/shared/widgets/texts/shared_texts.dart';
+import 'package:givt_app/shared/widgets/buttons/givt_elevated_button.dart';
+
+class AddMemberCounterPage extends StatefulWidget {
```
Comment (TammiLion) [inline lib/features/children/add_member/pages/add_member_counter_page.dart:10]: As Borat would say: Berry nice!

**Review by TammiLion — APPROVED** [review]
Just 1 question to look at.

File: lib/features/children/overview/pages/family_overview_page.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/children/overview/pages/family_overview_page.dart:103]: indeed that was to see it

**Pull request merged**